pkg/agent/execbridge.go

```diff
@@ -51,6 +51,29 @@ func (iface *execPluginInterface) Request(
 		iface.runner.recordResourceChange(*lastPermit, resp.Permit, iface.runner.global.metrics.schedulerApprovedChange)
 	}
 
+	successful := func() bool {
+		if err != nil {
+			return false
+		}
+		if resp.Permit == target {
+			return true
+		}
+		if lastPermit != nil {
+			if resp.Permit == *lastPermit {
+				return false
+			}
+		}
+		return true
+	}()
+	iface.runner.status.update(iface.runner.global, func(ps podStatus) podStatus {
+		if successful {
```

Comment:
I didn't exactly understand your proposed solution. I see how this could be problematic in languages like C++ but afaik this is a common pattern in Go.

Reply:
I see. I rather tend to keep the shorter version, which is more concise and consistent with the rest of code.